jni/src/common/Foreign/JNI/Unsafe.hs

```diff
@@ -286,6 +297,16 @@ data JNIError = JNIError Prelude.String Int32
 
 instance Exception JNIError
 
+data VerboseNoSuchMethodException = VerboseNoSuchMethodException JNI.String JNI.String [JNI.String]
```

Comment:
This would have to include the method signature used for the lookup as well.

Probably a shorter name like`NoSuchMethod` would work.